src/octree/mod.rs

```diff
@@ -134,57 +139,89 @@ pub struct NodeData {
+        let deprecated_min = bounding_box.deprecated_min.clone().unwrap(); // Version 9
+        let mut v = proto::Vector3d::new();
+        v.set_x(f64::from(deprecated_min.x));
+        v.set_y(f64::from(deprecated_min.y));
+        v.set_z(f64::from(deprecated_min.z));
+        v
+    });
+    let max = bounding_box.max.clone().unwrap_or_else(|| {
+        let deprecated_max = bounding_box.deprecated_max.clone().unwrap(); // Version 9
+        let mut v = proto::Vector3d::new();
+        v.set_x(f64::from(deprecated_max.x));
+        v.set_y(f64::from(deprecated_max.y));
+        v.set_z(f64::from(deprecated_max.z));
+        v
+    });
+    Aabb3::new(
+        Point3::new(min.x, min.y, min.z),
+        Point3::new(max.x, max.y, max.z),
+    )
+}
+
 impl Octree {
     // TODO(sirver): This creates an object that is only partially usable.
     pub fn from_data_provider(data_provider: Box<dyn DataProvider>) -> Result<Self> {
+        let bounding_box;
+        let meta;
+        let nodes_proto;
+
         let meta_proto = data_provider.meta_proto()?;
         match meta_proto.version {
-            9 | 10 => println!(
-                "Data is an older octree version: {}, current would be {}. \
-                 If feasible, try upgrading this octree using `upgrade_octree`.",
-                meta_proto.version, CURRENT_VERSION
-            ),
-            CURRENT_VERSION => (),
+            9 | 10 => {
+                println!(
+                    "Data is an older octree version: {}, current would be {}. \
+                     If feasible, try upgrading this octree using `upgrade_octree`.",
+                    meta_proto.version, CURRENT_VERSION
+                );
+                bounding_box = bounding_box_to_aabb(meta_proto.get_deprecated_bounding_box());
+                meta = OctreeMeta {
+                    resolution: meta_proto.deprecated_resolution,
+                    bounding_box,
+                };
+                nodes_proto = meta_proto.get_deprecated_nodes();
+            }
+            11 => {
+                println!(
+                    "Data is an older octree version: {}, current would be {}. \
+                     If feasible, try upgrading this octree using `upgrade_octree`.",
+                    meta_proto.version, CURRENT_VERSION
+                );
+                bounding_box = bounding_box_to_aabb(&meta_proto.get_deprecated_bounding_box());
+                meta = OctreeMeta {
+                    resolution: meta_proto.deprecated_resolution,
+                    bounding_box,
+                };
+                nodes_proto = meta_proto.get_deprecated_nodes();
+            }
+            CURRENT_VERSION => {
+                if !meta_proto.has_octree() {
+                    return Err(ErrorKind::InvalidInput("No octree meta found".to_string()).into());
+                }
+                let octree_meta = Some(meta_proto.get_octree()).unwrap();
```

Comment:
`Some(x).unwrap()` – probably left over from refactoring? :)

Reply:
no, it's intentional. you are left with a sort of 'pointer', you have to unwrap it to access the structure within